src/EventSubscriber/EzcontentJsonapiSubscriber.php

```diff
@@ -61,10 +76,31 @@ public function onResponse(FilterResponseEvent $event) {
     }
   }
 
+  /**
+   * Add cache context to jsonapi routes.
+   *
+   * Unpublished nodes are made accessible using the token generated by
+   * access_unpublished module. This token is passed as a query parameter in
+   * the jsonapi url whose cache-context needs to be added for cache
+   * invalidating.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\FilterResponseEvent $event
+   *   The event to process.
+   */
+  public function onRequest(Event $event) {
+    // Invalidate cache for content listing block for jsonapi resource.
+    $entity = $this->routeMatch->getParameter('entity');
+    if($entity instanceof BlockContent && $entity->bundle() === 'content_listing_component'
+      && $this->routeMatch->getRouteName() === 'jsonapi.block_content--content_listing_component.individual') {
+      Cache::invalidateTags($entity->getCacheTags());
+    }
+  }
+
```

Comment:
I think we could do this in the request event itself. If not, lets update the docblock for the method to state the reason on why we are invalidating cache in request event.